server/controllers/insertFood/index.js

```diff
@@ -0,0 +1,35 @@
+const Boom = require('@hapi/boom');
+const { insertFoodQuery } = require('../../database/queries');
+const { insertFoodValidation } = require('../../utils');
+
+const insertFoodController = async (req, res, next) => {
+  const { grams } = req.body;
+  const { id: userId } = req.user;
+  const { categoryId, foodId } = req.params;
+  try {
+    try {
+      await insertFoodValidation.validateAsync(
+        {
+          userId,
+          foodId,
+          categoryId,
+          grams,
+        },
+        { abortEarly: false },
+      );
+    } catch (err) {
+      throw Boom.badRequest(err.details.map((e) => e.message).join('\n'));
+    }
+
+    if (categoryId > 4 || categoryId < 1) {
+      throw Boom.notFound('category not found!');
+    }
+
+    await insertFoodQuery(userId, foodId, categoryId, grams);
+    res.json({ message: 'Food Successfully Added', status: 200 });
```

Comment:
It can be helpful to have your POST apis return a data object containing all of the new item's properties to the client.  It might not be as useful here, but in the future it could be a useful pattern for you.
For example:
res.json({ message: 'Food Successfully Added', status: 200, data: {foodId, categoryId, grams} });

Then, if there were any data properties that the backend created for this food item (for example a "created_at" timestamp) then the client could know what they were.